esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2314,7 +2314,13 @@ void CFileSprayEx::getDropZoneInfoByIP(const char* ip, const char* destFileIn, S
 
     StringBuffer destFile;
     if (isAbsolutePath(destFileIn))
+    {
         destFile.set(destFileIn);
+        // if multiple DZs per computer, use umask of single DZ returned
+        dropZone->getUMask(maskBuf);
+        if (maskBuf.length())
+            mask.set(maskBuf.str());
```

Comment:
@wangkx this is where I force umask for DZ found so that if there is more than one DZ per host it is used.